src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
-        resultBuilder: ControlResultBuilder,
-    ): Promise<void> {
-        const { action, values } = unpackValueControlIntent((input.request as IntentRequest).intent);
-        const valueStr = values[0].slotValue;
-        this.setValue(valueStr);
-        await this.commonHandlerWhenValueChanged(action ?? $.Action.Set, input, resultBuilder);
+    private handleClearValue(input: ControlInput, resultBuilder: ControlResultBuilder) {
+        resultBuilder.addAct(
+            new ValueClearedAct(this, {
+                value: this.state.value,
+                renderedValue: this.props.valueRenderer(this.state.value, input),
+            }),
+        );
+        this.clear();
+        return;
     }
 
-    private isTargetsMatchWithoutFeedbackNorValueWhenConfirmingValue(input: ControlInput): boolean {
+    private isSelectChoiceByTouch(input: ControlInput): boolean {
         try {
-            okIf(InputUtil.isIntent(input, GeneralControlIntent.name));
-            const { feedback, action, target } = unpackGeneralControlIntent(
-                (input.request as IntentRequest).intent,
-            );
-            okIf(InputUtil.feedbackIsUndefined(feedback));
-            okIf(
-                InputUtil.actionIsMatchOrUndefined(action, [
-                    ...this.props.interactionModel.actions.set,
-                    ...this.props.interactionModel.actions.change,
-                ]),
-            );
-            okIf(
-                InputUtil.targetIsMatch(target, _.without(this.props.interactionModel.targets, $.Target.It)),
-            );
-            this.handleFunc = this.handleTargetMatchWithoutFeedbackNorValueWhenConfirmingValue;
+            okIf(InputUtil.isAPLUserEventWithMatchingControlId(input, this.id));
+            const userEvent = input.request as interfaces.alexa.presentation.apl.UserEvent;
+            okIf(userEvent.arguments !== undefined);
+            okIf(userEvent.arguments.length === 2);
+            const controlId = (input.request as interfaces.alexa.presentation.apl.UserEvent)
+                .arguments![0] as string;
+            okIf(controlId === this.id);
             return true;
         } catch (e) {
             return falseIfGuardFailed(e);
         }
     }
 
-    private handleTargetMatchWithoutFeedbackNorValueWhenConfirmingValue(
-        input: ControlInput,
-        resultBuilder: ControlResultBuilder,
-    ): void {
-        resultBuilder.addAct(new RequestValueAct(this));
+    private async handleSelectChoiceByTouch(input: ControlInput, resultBuilder: ControlResultBuilder) {
+        const valueStr = (input.request as interfaces.alexa.presentation.apl.UserEvent)
+            .arguments![1] as string;
+        this.setValue(valueStr);
+        this.state.confirmed = true;
+        await this.validateAndAddActs(input, resultBuilder, $.Action.Set);
+        return;
     }
 
-    async commonHandlerWhenValueChanged(
-        action: string,
+    async validateAndAddActs(
```

Comment:
name suggestion: validateAndAddCommonFeedbackActs

minor: given that this method is doing a few things, consider a refactor.. Its not clear to me whether there is something better

Reply:
Renamed to `validateAndAddCommonFeedbackActs`
This function does run a validation check on value in state and adds corresponding ValueSet/ValueChanged acts if input is valid or adds InvalidValueAct. 
so the func name should be fine right?

Reply:
yep new name is good.